kernel/mouse/src/lib.rs

```diff
@@ -56,18 +100,20 @@ pub fn init(mut mouse: PS2Mouse<'static>, mouse_queue_producer: Queue<Event>) ->
 /// 
 /// In some cases (e.g. on device init), [the PS/2 controller can also send an interrupt](https://wiki.osdev.org/%228042%22_PS/2_Controller#Interrupts).
 extern "x86-interrupt" fn ps2_mouse_handler(_stack_frame: InterruptStackFrame) {
-    if let Some(MouseInterruptParams { mouse, queue }) = MOUSE.get() {
+    if let Some(MouseInterruptParams { mouse, queue, packet_bytes }) = MOUSE.get() {
+        // using `while` here didn't interact well with the window manager and increases handler runtime
```

Comment:
I've tried using a loop here and will try again, since some of the code changed.
This would handle multi-byte interrupts.

Reply:
yeah, I recall the previous code there. I phrased my comment as a question because legitimately not sure if the PS/2 mouse device actually allows for multiple bytes to be read per interrupt (i.e., does it do "batching" of multiple bytes into a single interrupt?); once we answer that then we can determine what the best choice is here.

Reply:
Yes it does, I've seen some osdev wiki people use `while` here 👍